gnovm/stdlibs/testing/fuzz.gno

```diff
@@ -0,0 +1,259 @@
+package testing
+
+import (
+	"math"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/ufmt"
+)
+
+type Fuzzer interface {
+	InsertDeleteMutate(p float64) Fuzzer
+	Mutate() Fuzzer
+	String() string
+}
+
+type StringFuzzer struct {
+	Value string
+	f     *F
+}
+
+// Mutate changes a StringFuzzer's value by replacing a random character
+// with a random ASCII character.
+func (sf *StringFuzzer) Mutate() Fuzzer {
+	runes := []rune(sf.Value)
+	if len(runes) == 0 {
+		return sf
+	}
+	index := randRange(0, len(runes)-1)
+	runes[index] = randomASCIIChar()
+	return &StringFuzzer{Value: string(runes)}
+}
+
+func (sf *StringFuzzer) InsertDeleteMutate(p float64) Fuzzer {
+	return &StringFuzzer{Value: InsertDelete(sf.Value, p)}
+}
+
+func (sf *StringFuzzer) Fuzz() string {
```

Comment:
You could use less scopes so the code is left leaning and needs less brain power to read.